### DISCUSSION TITLE
FirewallD daemon exited (uncleanly) and left system without any firewall rules

### DISCUSSION THREAD

**Discussion by cobratbq** [firewalld/firewalld#1105] [Q&A]
Hi, I'm not sure whether this is a bug or an external issue, so I would like to discuss.

I discovered the `firewalld` service on my desktop was inactive, and checked to discover that also all the firewall rules were gone. (`sudo nft list ruleset`) This happened on a Debian system with a backported version `1.3.0` of `firewalld`. I did not find further details to the nature the premature service exit.

`journalctl -u firewalld`
```
<timestamp> <host> systemd[1]: firewalld.service: Main process exited, code=exited, status=3/NOTIMPLEMENTED
<timestamp> <host> systemd[1]: firewalld.service: Failed with result 'exit-code'.
```

Obviously it is not expected that the firewalld service just exits, but apart from that, I was surprised to find the rules cleaned up.

Could it be the case that  `firewalld` cannot distinguish between graceful and failure-based exits and cleans up the firewall rules irrespectively? (Should this be considered a bug?)
I am aware of `CleanupOnExit=yes`. Would change this config solve this exceptional case?

**Comment by erig0** [comment]
v1.3.0 includes the startup failsafe which falls back to the stock configuration. If that's failing, then it's likely the underlying firewall is broken.

You can try enabling debug in `/etc/sysconfig/firewalld`. Check the log file `/var/log/firewalld`.

**Reply by cobratbq** [reply]
I am not sure I understand the intentions above, so correct me if I am missing the point.

> [..] startup failsafe which falls back to the stock configuration.

So, the (unchanged) configuration worked after starting manually. Additionally, `firewalld` was stopped when I discovered it, i.e. process not running + systemd reporting status as stopped. The only indicator I have is that `firewalld` "stopped running" and when I checked the rules themselves were gone too. I do not think there was ever a need to use the fallback.

I will need to try with debugging enabled.
